bin/compile

```diff
@@ -36,16 +36,19 @@ log "Generate requirements.txt with Poetry"
 
 log "Install Poetry"
 
-curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py \
+curl -sSL https://install.python-poetry.org \
     | sed -e 's/allowed_executables = \["python", "python3"\]/allowed_executables = ["python3"]/' \
     | python3 \
     | indent
 
-PATH_TO_POETRY_ENV="$HOME/.poetry/env"
+log "Add poetry to the PATH"
+export PATH="/app/.local/bin:$PATH"
 
-if [ -f "$PATH_TO_POETRY_ENV" ] ; then
-    # shellcheck source=/dev/null
-    . "$PATH_TO_POETRY_ENV"
+if [ "${POETRY_VERSION:0:3}" = "1.2" ] ; then
+    # Allow `poetry export` to work even if we aren't using the
+    # Python version that the project will ultimately run in
+    log "Set virtualenvs.prefer-active-python to true"
+    poetry config virtualenvs.prefer-active-python true | indent
```

Comment:
Doesn't feel too great, but probably this is indeed the way to go. We are testing this on our infra right now.

Reply:
Agreed! Could probably implement it in different ways, but I’m not sure what else to do besides “use the Python currently available” and I don’t *think* the Python version you’re running affects the `poetry export` output? Didn’t totally confirm that myself though.

Reply:
No, it shouldn't. This is why I'm irritated that since Poetry 1.2 they decided to enforce the selection of the virtualenv to be able to run `export` in the first place. But maybe this has something to do with `export` now being offloaded into a plugin, and they didn't really think about the consequences.